src/lib/viewers/BaseViewer.js

```diff
@@ -623,27 +623,51 @@ class BaseViewer extends EventEmitter {
     //--------------------------------------------------------------------------
 
     /**
+     * Get the configuration for viewer annotations and transform if legacy format.
+     *
+     * @return {Object} An object containing configuration properties.
+     */
+    getViewerAnnotationsConfig() {
+        const config = this.getViewerOption('annotations') || {};
+
+        // Backwards compatability for old boolean flag usage
+        if (typeof config === 'boolean') {
+            return {
+                enabled: config
+            };
+        }
+
+        return config;
+    }
+
+    /**
      * Loads the appropriate annotator and loads the file's annotations
      *
      * @protected
      * @return {void}
      */
     loadAnnotator() {
+        // Do nothing if annotations are disabled for the viewer
+        if (!this.areAnnotationsEnabled()) {
+            return;
+        }
+
         /* global BoxAnnotations */
         const boxAnnotations = new BoxAnnotations();
+        // #TODO(@spramod|@jholdstock): remove this after we have annotation instancing
+        const viewerName = this.options.viewer.NAME;
+        const annotationsConfig = this.getViewerAnnotationsConfig();
 
-        this.annotatorConf = boxAnnotations.determineAnnotator(this.options.viewer.NAME);
+        this.annotatorConf = boxAnnotations.determineAnnotator(viewerName, undefined, annotationsConfig);
```

Comment:
See comment above but if we modify the function declaration, then we can call it like this
```boxAnnotations.determineAnnotator(viewerName, annotationsConfig);```